### ISSUE TITLE
ubuntu16.04-pcl-error:vcpkg/installed/x64-linux/share/vtk/../../lib/proj.lib

### ISSUE THREAD

**Issue by yangninghua** [microsoft/vcpkg#8718]
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.12)
project(PCL_demo1)

set(CMAKE_CXX_STANDARD 11)


find_package(PCL CONFIG REQUIRED)
#find_package(Boost REQUIRED)

include_directories(${PCL_INCLUDE_DIRS})
link_directories(${PCL_LIBRARY_DIRS})
add_definitions(${PCL_DEFINITIONS})
#list(REMOVE_ITEM PCL_LIBRARIES "vtkproj4")

add_executable(PCL_demo1 main.cpp)
target_link_libraries (PCL_demo1 ${PCL_LIBRARIES})
```

```
/home/spple/clion-2018.2.5/bin/cmake/linux/bin/cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_TOOLCHAIN_FILE=/home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake -G "CodeBlocks - Unix Makefiles" /home/spple/CLionProjects/vcpkg_PCL_demo1
-- Eigen found (include: /home/spple/paddle/vcpkg/installed/x64-linux/include/eigen3, version: 3.3.7)
CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


CMake Warning at /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:847 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:963 (_Boost_COMPONENT_DEPENDENCIES)
  /home/spple/clion-2018.2.5/bin/cmake/linux/share/cmake-3.12/Modules/FindBoost.cmake:1622 (_Boost_MISSING_DEPENDENCIES)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:227 (_find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:138 (find_package)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:302 (find_boost)
  /home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake:526 (find_external_library)
  /home/spple/paddle/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:7 (find_package)


-- Boost version: 1.71.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   thread
--   date_time
--   iostreams
--   serialization
--   chrono
--   atomic
--   regex
** WARNING ** io features related to png will be disabled
-- QHULL found (include: /home/spple/paddle/vcpkg/installed/x64-linux/include, lib: optimized;/home/spple/paddle/vcpkg/installed/x64-linux/lib/libqhullstatic.a;debug;/home/spple/paddle/vcpkg/installed/x64-linux/debug/lib/libqhullstatic_d.a)
-- looking for PCL_COMMON
-- looking for PCL_KDTREE
-- looking for PCL_OCTREE
-- looking for PCL_SEARCH
-- looking for PCL_SAMPLE_CONSENSUS
-- looking for PCL_FILTERS
-- looking for PCL_2D
-- looking for PCL_GEOMETRY
-- looking for PCL_IO
-- looking for PCL_FEATURES
-- looking for PCL_ML
-- looking for PCL_SEGMENTATION
-- looking for PCL_VISUALIZATION
-- looking for PCL_SURFACE
-- looking for PCL_REGISTRATION
-- looking for PCL_KEYPOINTS
-- looking for PCL_TRACKING
-- looking for PCL_RECOGNITION
-- looking for PCL_STEREO
-- looking for PCL_OUTOFCORE
-- looking for PCL_PEOPLE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/spple/CLionProjects/vcpkg_PCL_demo1/cmake-build-debug

[Finished]
```

build-error
```
/home/spple/clion-2018.2.5/bin/cmake/linux/bin/cmake --build /home/spple/CLionProjects/vcpkg_PCL_demo1/cmake-build-debug --target PCL_demo1 -- -j 6
Scanning dependencies of target PCL_demo1
[ 50%] Building CXX object CMakeFiles/PCL_demo1.dir/main.cpp.o
make[3]: *** No rule to make target '/home/spple/paddle/vcpkg/installed/x64-linux/share/vtk/../../lib/proj.lib', needed by 'PCL_demo1'。 停止。
make[3]: *** 正在等待未完成的任务....
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/PCL_demo1.dir/all' failed
make[2]: *** [CMakeFiles/PCL_demo1.dir/all] Error 2
CMakeFiles/Makefile2:84: recipe for target 'CMakeFiles/PCL_demo1.dir/rule' failed
make[1]: *** [CMakeFiles/PCL_demo1.dir/rule] Error 2
Makefile:118: recipe for target 'PCL_demo1' failed
make: *** [PCL_demo1] Error 2
```

**Comment by JackBoosY** [comment]
@yangninghua Thanks for reporting this issue.
Could you provide _/home/spple/paddle/vcpkg/installed/x64-linux/share/pcl/PCLConfig.cmake_ ?
Thanks.

**Comment by yangninghua** [comment]
```
#-----------------------------------------------------------------------------
#
# VTKConfig.cmake - VTK CMake configuration file for external projects.
#
# This file is configured by VTK and used by the UseVTK.cmake module
# to load VTK's settings for an external project.

# Compute the installation prefix from this VTKConfig.cmake file location.
set(_vtk_installed_prefix "${CMAKE_CURRENT_LIST_DIR}/../../share/vtk")
set(_vtk_requested_prefix "${CMAKE_CURRENT_LIST_DIR}")
get_filename_component(_vtk_installed_prefix_full "${_vtk_installed_prefix}" REALPATH)
get_filename_component(_vtk_requested_prefix_full "${_vtk_requested_prefix}" REALPATH)
if (_vtk_installed_prefix_full STREQUAL _vtk_requested_prefix_full)
  set(VTK_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../..")
else ()
  set(VTK_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}")
  get_filename_component(VTK_INSTALL_PREFIX "${VTK_INSTALL_PREFIX}" PATH)
  get_filename_component(VTK_INSTALL_PREFIX "${VTK_INSTALL_PREFIX}" PATH)
endif ()
set(VTK_MODULES_DIR "${VTK_INSTALL_PREFIX}/share/vtk/Modules")

set (__vtk_install_tree TRUE)

if (CMAKE_VERSION VERSION_LESS "3.3")
    message(FATAL_ERROR "VTK now requires CMake 3.3 or newer")
  endif()

#-----------------------------------------------------------------------------
# Minimum compiler version check: GCC >= 4.6
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.6)
  message(FATAL_ERROR "GCC 4.6 or later is required.")
endif ()

#-----------------------------------------------------------------------------
# Minimum compiler version check: LLVM Clang >= 3.0
if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.0)
  message(FATAL_ERROR "LLVM Clang 3.0 or later is required.")
endif ()

#-----------------------------------------------------------------------------
# Minimum compiler version check: Apple Clang >= 3.0 (Xcode 4.2)
if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.0)
  message(FATAL_ERROR "Apple Clang 3.0 or later is required.")
endif ()

#-----------------------------------------------------------------------------
# Minimum compiler version check: Microsoft C/C++ >= 17.0 (aka VS 2012 aka VS 11.0)
if ("x${CMAKE_CXX_COMPILER_ID}" STREQUAL "xMSVC" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 17.0)
  message(FATAL_ERROR "Microsoft Visual Studio 2012 or later is required.")
endif ()

#-----------------------------------------------------------------------------
# Minimum compiler version check: Intel C++ (ICC) >= 14
if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS 14.0)
  message(FATAL_ERROR "Intel C++ (ICC) 14.0 or later is required.")
endif ()

# The C and C++ flags added by VTK to the cmake-configured flags.
set(VTK_REQUIRED_C_FLAGS "")
set(VTK_REQUIRED_CXX_FLAGS "")
set(VTK_REQUIRED_EXE_LINKER_FLAGS "")
set(VTK_REQUIRED_SHARED_LINKER_FLAGS "")
set(VTK_REQUIRED_MODULE_LINKER_FLAGS "")

# The VTK version number
set(VTK_MAJOR_VERSION "8")
set(VTK_MINOR_VERSION "2")
set(VTK_BUILD_VERSION "0")

# The location of the UseVTK.cmake file.
set(VTK_CMAKE_DIR "${VTK_INSTALL_PREFIX}/share/vtk")
set(VTK_USE_FILE "${VTK_CMAKE_DIR}/UseVTK.cmake")

# The rendering backend VTK was configured to use.
set(VTK_RENDERING_BACKEND "OpenGL2")

if (__vtk_install_tree)
  if (WIN32)
    set (VTK_RUNTIME_DIRS "${CMAKE_CURRENT_LIST_DIR}/../../bin")
  else ()
    set (VTK_RUNTIME_DIRS "${CMAKE_CURRENT_LIST_DIR}/../../lib")
  endif ()
else()
  if (WIN32)
    set (VTK_RUNTIME_DIRS "/media/spple/新加卷/Dataset/vcpkg-master/buildtrees/vtk/x64-linux-rel/bin")
  else ()
    set (VTK_RUNTIME_DIRS "/media/spple/新加卷/Dataset/vcpkg-master/buildtrees/vtk/x64-linux-rel/lib")
  endif ()
endif()

# Setup VTK-m if it was enabled
set(VTK_HAS_VTKM false)
if(VTK_HAS_VTKM AND NOT TARGET vtkm_cont)
  set(VTKM_CMAKE_DIR "${VTK_CMAKE_DIR}")
  if(NOT __vtk_install_tree)
    set(VTKM_CMAKE_DIR "/share/vtk/vtkm")
  endif()
  get_filename_component(VTKM_CMAKE_DIR "${VTKM_CMAKE_DIR}" PATH)
  find_package(VTKm
               PATHS "${CMAKE_CURRENT_LIST_DIR}"
                     "${VTKM_CMAKE_DIR}"
                     "${VTK_RUNTIME_DIRS}"
               NO_DEFAULT_PATH
               )
endif()


#-----------------------------------------------------------------------------
# Load requested modules.

# List of available VTK modules.
set(VTK_MODULES_ENABLED "vtkkwiml;vtksys;vtkutf8;vtkCommonCore;vtkCommonMath;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkCommonDataModel;vtkCommonColor;vtkCommonExecutionModel;vtkCommonComputationalGeometry;vtkFiltersCore;vtkFiltersGeneral;vtkImagingCore;vtkImagingFourier;vtkeigen;vtkFiltersStatistics;vtkFiltersExtraction;vtkInfovisCore;vtkFiltersGeometry;vtkFiltersSources;vtkRenderingCore;vtkzlib;vtkfreetype;vtkRenderingFreeType;vtkRenderingContext2D;vtkChartsCore;vtkDICOMParser;vtkdoubleconversion;vtklz4;vtklzma;vtkIOCore;vtkIOLegacy;vtkexpat;vtkIOXMLParser;vtkDomainsChemistry;vtkglew;vtkRenderingOpenGL2;vtkDomainsChemistryOpenGL2;vtkIOXML;vtkParallelCore;vtkFiltersAMR;vtkFiltersFlowPaths;vtkFiltersGeneric;vtkImagingSources;vtkFiltersHybrid;vtkFiltersHyperTree;vtkImagingGeneral;vtkFiltersImaging;vtkFiltersModeling;vtkFiltersParallel;vtkFiltersParallelImaging;vtkFiltersPoints;vtkFiltersProgrammable;vtkFiltersSMP;vtkFiltersSelection;vtkFiltersTexture;vtkFiltersTopology;vtkverdict;vtkFiltersVerdict;vtkMetaIO;vtkjpeg;vtkpng;vtktiff;vtkIOImage;vtkImagingHybrid;vtkInfovisLayout;vtkInteractionStyle;vtkImagingColor;vtkRenderingAnnotation;vtkRenderingVolume;vtkInteractionWidgets;vtkViewsCore;vtklibproj;vtkGeovisCore;vtkhdf5;vtkIOAMR;vtkIOAsynchronous;vtkpugixml;vtkIOCityGML;vtkIOEnSight;vtknetcdf;vtkexodusII;vtkIOExodus;vtkgl2ps;vtkRenderingGL2PSOpenGL2;vtkIOExport;vtkIOExportOpenGL2;vtklibharu;vtkIOExportPDF;vtkIOGeometry;vtkIOImport;vtklibxml2;vtkIOInfovis;vtkIOLSDyna;vtkIOMINC;vtkogg;vtktheora;vtkIOMovie;vtkIONetCDF;vtkIOPLY;vtkjsoncpp;vtkIOParallel;vtkIOParallelXML;vtksqlite;vtkIOSQL;vtkIOSegY;vtkIOTecplotTable;vtkIOVeraOut;vtkIOVideo;vtkImagingMath;vtkImagingMorphological;vtkImagingStatistics;vtkImagingStencil;vtkInteractionImage;vtkRenderingContextOpenGL2;vtkRenderingImage;vtkRenderingLOD;vtkRenderingLabel;vtkRenderingVolumeOpenGL2;vtkViewsContext2D;vtkViewsInfovis;vtkViewsGeovis")

# Import VTK targets.
set(VTK_CONFIG_TARGETS_FILE "${VTK_INSTALL_PREFIX}/share/vtk/VTKTargets.cmake")
if(NOT TARGET vtkCommonCore)
  include("${VTK_CONFIG_TARGETS_FILE}")
endif()

# Load module interface macros.
include("${VTK_INSTALL_PREFIX}/share/vtk/vtkModuleAPI.cmake")

# Compute set of requested modules.
if(VTK_FIND_COMPONENTS)
  # Specific modules requested by find_package(VTK).
  set(VTK_MODULES_REQUESTED "${VTK_FIND_COMPONENTS}")
else()
  # No specific modules requested.  Use all of them.
  set(VTK_MODULES_REQUESTED "${VTK_MODULES_ENABLED}")
endif()

# Load requested modules and their dependencies into variables:
#  VTK_DEFINITIONS     = Preprocessor definitions
#  VTK_LIBRARIES       = Libraries to link
#  VTK_INCLUDE_DIRS    = Header file search path
#  VTK_LIBRARY_DIRS    = Library search path (for outside dependencies)
vtk_module_config(VTK ${VTK_MODULES_REQUESTED})

#-----------------------------------------------------------------------------

# VTK global configuration options.
set(VTK_BUILD_SHARED_LIBS "OFF")
set(VTK_LEGACY_REMOVE "OFF")
set(VTK_LEGACY_SILENT "OFF")
set(VTK_WRAP_PYTHON "OFF")
set(VTK_WRAP_JAVA "OFF")
set(VTK_QT_VERSION "")
set(VTK_ENABLE_KITS "OFF")

# Do not add options or information here that is specific to a
# particular module.  Instead set <module>_EXPORT_OPTIONS and/or
# <module>_EXPORT_CODE_BUILD and <module>_EXPORT_CODE_INSTALL
# at the top of the module CMakeLists.txt file.
```

**Comment by yangninghua** [comment]
new build
```
/home/boyun/software/clion-2018.3.4/bin/cmake/linux/bin/cmake --build /home/boyun/CLionProjects/PCL_demo1/cmake-build-debug --target PCL_demo1 -- -j 4
Scanning dependencies of target PCL_demo1
[ 50%] Building CXX object CMakeFiles/PCL_demo1.dir/main.cpp.o
make[3]: *** No rule to make target '/media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/share/vtk/../../lib/proj.lib', needed by 'PCL_demo1'。 停止。
make[3]: *** 正在等待未完成的任务....
In file included from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/boost/detail/endian.hpp:9:0,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/PCLPointCloud2.h:11,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/pcl_base.h:55,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/common/common.h:41,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/common/common_headers.h:39,
                 from /home/boyun/CLionProjects/PCL_demo1/main.cpp:2:
/media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/boost/predef/detail/endian_compat.h:11:161: note: #pragma message: The use of BOOST_*_ENDIAN and BOOST_BYTE_ORDER is deprecated. Please include <boost/predef/other/endian.h> and use BOOST_ENDIAN_*_BYTE instead
 #pragma message("The use of BOOST_*_ENDIAN and BOOST_BYTE_ORDER is deprecated. Please include <boost/predef/other/endian.h> and use BOOST_ENDIAN_*_BYTE instead")
                                                                                                                                                                 ^
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/PCL_demo1.dir/all' failed
make[2]: *** [CMakeFiles/PCL_demo1.dir/all] Error 2
CMakeFiles/Makefile2:84: recipe for target 'CMakeFiles/PCL_demo1.dir/rule' failed
make[1]: *** [CMakeFiles/PCL_demo1.dir/rule] Error 2
Makefile:118: recipe for target 'PCL_demo1' failed
make: *** [PCL_demo1] Error 2
```

**Comment by yangninghua** [comment]
Release
```
/home/boyun/software/clion-2018.3.4/bin/cmake/linux/bin/cmake --build /home/boyun/CLionProjects/PCL_demo1/cmake-build-release --target PCL_demo1 -- -j 4
Scanning dependencies of target PCL_demo1
[ 50%] Building CXX object CMakeFiles/PCL_demo1.dir/main.cpp.o
make[3]: *** No rule to make target '/media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/share/vtk/../../lib/proj.lib', needed by 'PCL_demo1'。 停止。
make[3]: *** 正在等待未完成的任务....
In file included from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/boost/detail/endian.hpp:9:0,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/PCLPointCloud2.h:11,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/pcl_base.h:55,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/common/common.h:41,
                 from /media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/pcl/common/common_headers.h:39,
                 from /home/boyun/CLionProjects/PCL_demo1/main.cpp:2:
/media/boyun/NO_1/ubuntu16.04/vcpkg/vcpkg/installed/x64-linux/include/boost/predef/detail/endian_compat.h:11:161: note: #pragma message: The use of BOOST_*_ENDIAN and BOOST_BYTE_ORDER is deprecated. Please include <boost/predef/other/endian.h> and use BOOST_ENDIAN_*_BYTE instead
 #pragma message("The use of BOOST_*_ENDIAN and BOOST_BYTE_ORDER is deprecated. Please include <boost/predef/other/endian.h> and use BOOST_ENDIAN_*_BYTE instead")
                                                                                                                                                                 ^
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/PCL_demo1.dir/all' failed
make[2]: *** [CMakeFiles/PCL_demo1.dir/all] Error 2
CMakeFiles/Makefile2:84: recipe for target 'CMakeFiles/PCL_demo1.dir/rule' failed
make[1]: *** [CMakeFiles/PCL_demo1.dir/rule] Error 2
Makefile:118: recipe for target 'PCL_demo1' failed
make: *** [PCL_demo1] Error 2
```

**Comment by ISWBenjaminKaiser** [comment]
Hello, I have the same issue. Were you be able to fix it yet?

**Comment by JackBoosY** [comment]
Modify _VCPKG_PATH/installed/x64-linux/share/vtk/Modules/vtklibproj.cmake_, change `proj.lib` to `libproj.a` can solve this issue.

**Comment by ISWBenjaminKaiser** [comment]
Thank you for your reply.
It works but i had to change it in VCPKG_PATH/installed/x64-linux/share/vtk/VTKTargets.cmake as well!

**Comment by JackBoosY** [comment]
@ISWBenjaminKaiser Please use #9899 changes.

**Comment by JackBoosY** [comment]
Hi everyone, I've fixed this issue in #9899, please update vcpkg and rebuild vtk.

Thanks.
